jaxlib/cuda_lu_pivot.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import operator
+
+import numpy as np
+
+from jaxlib import xla_client
+
+try:
+  from . import cuda_lu_pivot_kernels
+  for _name, _value in cuda_lu_pivot_kernels.registrations().items():
+    xla_client.register_custom_call_target(_name, _value, platform="CUDA")
+except ImportError:
+  pass
+
+_prod = lambda xs: functools.reduce(operator.mul, xs, 1)
+
+
+# TODO(phawkins): remove after we no longer need to support old jax releases.
```

Comment:
You don't need this backward compatibility code. This function is always the identity in sufficiently new `jax` releases.